mountpoint-s3/scripts/network_performance.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+result_file=network_performance.json
+buffer=/tmp/buffer
+
+query_ec2_instance_network_throughput="aws ec2 describe-instance-types
+--filters \"Name=instance-type,Values=*\"
+--query \"InstanceTypes[].[InstanceType, NetworkInfo.NetworkPerformance]\"
+--output json"
+
+(> ${buffer})
+
+generate_json_entry() {
+    echo "{"
+    eval $query_ec2_instance_network_throughput | \
+        jq -c '.[]' | \
+        while read line;do
+            # extract instance type as a string. eg:"m5dn.12xlarge"
+            instance_type=`echo $line | cut -d ',' -f1 | sed 's/\[//'`
+
+            # extract instance throughput as float number. eg: 50
+            instance_throughput=`echo $line | cut -d ',' -f2 | sed 's/\s*[a-zA-Z]\s*//g' | sed 's/\]//' | sed 's/\"//g'`
+
+            # concate item as a string. eg: "m5dn.12xlarge":50
+            type_throughput_pair="$instance_type:$instance_throughput"
+
+            # some instance types is missing throughput numbers, appending null to make it a legal json element.
+            [[ $type_throughput_pair =~ .*:$ ]] && echo $type_throughput_pair'null,' >> ${buffer} || echo $type_throughput_pair',' >> ${buffer}
```

Comment:
I don't think it's necessary to out put error message to stderr, since those type of instances will be either use default throughput or customer specified throughput

Reply:
I think its worth logging it because it'll tell the person running the script (likely maintainers) when it encounters an instance type where we couldn't parse the throughput.

This may lead us to identify changes we need to make to the script.